pkg/controllers/csi/metadata/migrations.go

```diff
@@ -0,0 +1,93 @@
+package metadata
+
+import (
+	"path/filepath"
+
+	dtcsi "github.com/Dynatrace/dynatrace-operator/pkg/controllers/csi"
+	"gorm.io/gorm"
+)
+
+func dataMigration(tx *gorm.DB) error {
+	var dynakubes []Dynakube
+
+	tx.Table("dynakubes").Find(&dynakubes)
+
+	for _, d := range dynakubes {
+		tc := TenantConfig{
+			Name:                        d.Name,
+			TenantUUID:                  d.TenantUUID,
+			ConfigDirPath:               filepath.Join(filepath.Join(dtcsi.DataPath, d.TenantUUID), d.Name, dtcsi.SharedAgentConfigDir),
+			DownloadedCodeModuleVersion: d.LatestVersion,
+			MaxFailedMountAttempts:      int64(d.MaxFailedMountAttempts),
+		}
+
+		result := tx.Create(&tc)
+		if result.Error != nil {
+			log.Error(result.Error, "failed to create TenantConfig")
+		}
+
+		cm := CodeModule{
+			Version:  d.LatestVersion,
```

Comment:
same as above, it can be the `ImageDigest`

Reply:
fixed in 2639aa76